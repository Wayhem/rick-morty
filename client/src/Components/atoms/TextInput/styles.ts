import styled from 'styled-components'

export const TextInputContainer = styled.div`
  position: relative;
`

export const TextInput = styled.input<any>`
  display: inline-block;
  width: 500px;
  height: 40px;
  box-sizing: border-box;
  outline: none;
  border: 1px solid lightgray;
  border-radius: 3px;
  padding: 10px 10px 10px 100px;
  transition: all 0.1s ease-out;

  &:focus {
    padding: 10px;
    transition: all 0.3s ease-out;
    transition-delay: 0.2s;
  }

  & + label {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    height: 40px;
    line-height: 40px;
    color: white;
    border-radius: 3px 0 0 3px;
    width: 90px;
    background: ${props => props.theme.colors.orange};
    transform: translateZ(0) translateX(0);
    transition: all 0.3s ease-in;
    transition-delay: 0.2s;
    text-align: center;
  }

  &:focus + label {
    transform: translateY(${props => props.coordinateY}) translateX(${props => props.coordinateX});
    border-radius: 3px;
    transition: all 0.1s ease-out;
  }
`
