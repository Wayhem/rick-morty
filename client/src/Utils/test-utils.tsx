import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult, Queries } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { MemoryRouter as Router } from 'react-router-dom'
import { store } from 'Store'
import { theme } from 'Config/styled'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AllTheProviders: FC = ({ children }: { children?: any }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  </ThemeProvider>
)

AllTheProviders.defaultProps = {
  children: <></>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult<Queries, HTMLElement> => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { customRender as render }
