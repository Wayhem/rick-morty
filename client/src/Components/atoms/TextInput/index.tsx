import React from 'react'
import { TextInput, TextInputContainer } from 'Components/atoms/TextInput/styles'
import { positionShift } from 'Config/constants/inputLabelShift'

interface InputProps {
  placeholder: string;
  label: string;
  inputId: string;
  value: string;
  labelShift?: positionShift;
  type?: string;
  required?: boolean | undefined;
  onChange(e: any): void;
}

const Input = ({ onChange, placeholder, label, inputId, value, labelShift, type, required }: InputProps) => {
  const getCoordinates = (position: positionShift | undefined): { coordinateX: string; coordinateY: string } => {
    switch (position) {
      case positionShift.up:
        return { coordinateX: '0', coordinateY: '-120%' }
      case positionShift.left:
        return { coordinateX: '-120%', coordinateY: '0' }
      case positionShift.down:
        return { coordinateX: '0', coordinateY: '120%' }
      default:
        return { coordinateX: '-120%', coordinateY: '0' }
    }
  }

  const { coordinateX, coordinateY } = getCoordinates(labelShift)

  return (
    <TextInputContainer>
      <TextInput
        type={type}
        onChange={onChange}
        placeholder={placeholder}
        id={inputId}
        value={value}
        coordinateX={coordinateX}
        coordinateY={coordinateY}
        required={required}
      />
      <label htmlFor={inputId}>{label}</label>
    </TextInputContainer>
  )
}

Input.defaultProps = {
  labelShift: positionShift.left,
  type: 'text',
  required: undefined
}

export default Input
