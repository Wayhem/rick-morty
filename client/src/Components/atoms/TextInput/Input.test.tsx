import React from 'react'
import TextInput from 'Components/atoms/TextInput'
import { render } from 'Utils/test-utils'

test('TextInput renders and matches snapshot', () => {
  const mockProps = {
    placeholder: '',
    label: '',
    inputId: '',
    value: '',
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    onChange: () => {}
  }

  const { container } = render(<TextInput {...mockProps} />)
  expect(container).toMatchSnapshot()
})
