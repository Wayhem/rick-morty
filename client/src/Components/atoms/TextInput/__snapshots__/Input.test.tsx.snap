// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ VfWiw"
  >
    <input
      class="sc-gsTCUz gPDLux"
      id=""
      placeholder=""
      type="text"
      value=""
    />
    <label
      for=""
    />
  </div>
</div>
`;
